blog-posts/takeuntil-takewhile/takeWhile-takeUntil-takeWhat.md

```diff
@@ -0,0 +1,106 @@
+This way of handling subscription is quite imperative and therefore seems to be easier for people to understand.
+
+## About TakeUntil
+
+```ts
+@Component({...})
+export class MyComponent implements OnInit, OnDestroy {
+
+  private destroyed = new Subject();
+
+  public ngOnInit() {
+    myAwesomeObservable$.pipe(
+      takeUntil(this.destroyed$)
+    ).subscribe();
+  }
+
+  public ngOnDestroy() {
+    this.destroyed.next();
+    this.destroyed.complete();
+  }
+}
+```
+
+From an high-level perspective the code snippets don't look that different. Instead of having a boolean property, describing the state of our observable, we now use a `Subject`.
+We pass the `Subject` instance to the `takeUntil` operator and as soon as we want to unsubscribe from our Observable we emit a notification over our `Subject` with the `next` method. The call of the `complete` method isn't necessary for this example, but it is a good practice within the `takeUntil`-Pattern.
+
+## The Problem
+
+So all in all both code snippets look similar and behave similar, right? Well, No!
+Let's have a look at the marble diagrams describing those operators because this will highlight the difference between those two operators.
+
+![takeUntil marble diagram, kindly provided by Michael Hladky](./assets/takeUntil.png)
+_takeUntil marble diagram, kindly provided by [Michael Hladky](https://twitter.com/Michael_Hladky)_
+
+![takeWhile marble diagram, kindly provided by Michael Hladky](./assets/takeWhile.png)
+_takeWhile marble diagram, kindly provided by [Michael Hladky](https://twitter.com/Michael_Hladky)_
+
+The problem here is that `takeWhile` is intended to take an incoming notification, use some comparison on it, which might lead to an unsubscribe. The important fact is, that `takeWhile` is triggered by the incoming notification and might unsubscribe afterwards. `TakeUntil` in contrast is triggered by the passed Observable.
+That's why `takeWhile` might cause several issues. So definitely, it needs a new notification to unsubscribe. Imagine having a long-living Observable. You will need one notification more with `takeWhile` than with `takeUntil`. Also, this additional notification can initiate multiple processes within your observable. Imagine having code like this:
+
+```ts
+longLivingObservable$
+  .pipe(
+    tap(() => this.startAnimation()),
+    switchMap(val => this.makeHttpCall(val)),
+    takeWhile(val => this.alive),
+  )
+  .subscribe();
+```
+
+So what's the issue with this piece of code? Well, our component is already destroyed and due to the needed notification, needed before unsubscribing kicks in, we will start an animation and trigger an HTTP call. This is probably unwanted and just afterwards we will check if we want to unsubscribe from our Observable. Besides the fact that those operations are totally superfluous, it also might break our app or pollute our state.
```

Comment:
> So what's the issue with this piece of code? Well, our component is already destroyed and due to the needed notification, needed before unsubscribing kicks in,

This last part (`needed before unsubscribing kicks in`) reads kind of weird. I suppose you mean that is problematic because the `takeWhile` is at the end of th chain, meaning that it will still trigger the animation and http call?

Reply:
You could even say that it would cause a memory leak if the the `longLivingObservable$` never emits (nor completes) after the component is destroyed